### PR TITLE
Rename platform to application, adjustments for Experimenter API v6

### PR DESCRIPTION
**Pull request by k88hudson**
I noticed a few things that need to be updated with some of the renaming described here https://docs.google.com/document/d/16ekd-cXkJF8_UTgkZ3ufzJFusYz9htd5EE2NpIFFJp4/edit#

### CODE REVIEW

File: types/experiments.ts [hunk 1]
```diff
@@ -28,28 +28,28 @@ export interface Experiment {
   isEnrollmentPaused: boolean;
   /** Bucketing configuration */
   bucketConfig: BucketConfig;
-  /** A list of feature slugs relevant to the experiment analysis */
-  features: Array<string>;
+  /** A list of probe set slugs relevant to the experiment analysis */
+  probeSets: Array<string>;
```
Comment (k88hudson) [inline types/experiments.ts:32]: cc @tdsmith

**Comment by tdsmith** [comment]
ty! Jetstream will need a change after Experimenter starts emitting this name but I don't think it'll break anything, so please land with abandon.

**Review by jaredlockhart — COMMENTED** [review]
- Add channel (app_id on mobile)

**Review by jaredlockhart — CHANGES_REQUESTED** [review]
* Add channel (app_id on mobile)
* Flatten arguments up into the top level

**Review by jaredlockhart — CHANGES_REQUESTED** [review]
* Remove active
* remove all constants (change to just their types, so we only have constnats hard coded in one place ie experimenter)

**Review by jaredlockhart — APPROVED** [review]
👍 🐶

File: types/experiments.ts [hunk 2]
```diff
@@ -39,15 +37,16 @@ export interface NimbusExperiment {
...
+  /** Which channels should the application match? */
+  channels: Array<string>;
 }
 
 interface BucketConfig {
   /**
-   * The randomization unit. Note that client_id is not yet implemented.
-   * @default "normandy_id"
+   * The randomization unit.
    */
```
Comment (pdehaan) [inline types/experiments.ts:48]: We can possibly remove this comment since it doesn't really clarify/add anything.

Comment (k88hudson) [reply types/experiments.ts:48]: Oops, good point. I should probably revise the comments in this doc too.

Comment (pdehaan) [reply types/experiments.ts:48]: I did a quick check and don't _think_ we need to revise anything on https://mozilla.github.io/nimbus-shared/.

**Pull request merged**